src/Arcus.WebApi.Security/Authorization/Jwt/JwtTokenReader.cs

```diff
@@ -39,21 +44,33 @@ public JwtTokenReader(string applicationId)
         /// </summary>
         /// <remarks>Uses Microsoft OpenId connect discovery endpoint</remarks>
         /// <param name="tokenValidationParameters">Collection of parameters to influence how the token validation is done</param>
-        public JwtTokenReader(TokenValidationParameters tokenValidationParameters) : this(tokenValidationParameters, MicrosoftDiscoveryEndpoint)
+        /// <param name="claimCheck">Custom claims key-value pair to validate against</param>
+        public JwtTokenReader(TokenValidationParameters tokenValidationParameters, IDictionary<string, string> claimCheck = null) : this(tokenValidationParameters, MicrosoftDiscoveryEndpoint, claimCheck)
+        {
+        }
+
+        /// <summary>
+        ///     Constructor
+        /// </summary>
+        /// <param name="applicationId">Azure AD Application used as audience to validate against</param>
+        public JwtTokenReader(string applicationId) : this(new Dictionary<string, string> {{ AudienceKeyName, applicationId}})
         {
+            Guard.NotNullOrWhitespace(applicationId, nameof(applicationId));
```

Comment:
If you Guard on empty claimchecks (claimchecks where the value is not specified) in the other constructor, then this Guard here can be removed I guess

Reply:
Beyond validation, what I would like to verify with you at this point is whether a claim can be empty or not. If it is verified in the list of claims that none should be empty, it is assumed that the logic does not allow any claim with an empty or null value. What I do not know is whether in a real scenario this situation can occur